### DISCUSSION TITLE
Custom Certificate

### DISCUSSION THREAD

**Discussion by 0dayoverflow** [community/community#120698] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Hi. I'm trying to use copilot and sends this message: Im having a personal account how can I fix this? Thanks
![Captura de pantalla 2024-04-22 022513](https://github.com/community/community/assets/98193746/dd19894d-c857-4399-8d03-b2b2d49014de)

**Comment by althabitifs** [comment]
> ### Select Topic Area
> 
> Bug
> 
> ### Body
> 
> Hi. I'm trying to use copilot and sends this message: Im having a personal account how can I fix this? Thanks
> 
> ![Captura de pantalla 2024-04-22 022513](https://github.com/community/community/assets/98193746/dd19894d-c857-4399-8d03-b2b2d49014de)
> 
> 
> 
>

**Reply by 0dayoverflow** [reply]
so what it sthe solution?

**Reply by ayim** [reply]
@0dayoverflow you could upgrade to Copilot for Business, or alternatively, you could try this workaround:
https://stackoverflow.com/questions/75898544/visual-studio-2022-coplit-showyour-current-copilot-license-doesnt-support-prox

**Reply by 0dayoverflow** [reply]
And how do i install the pem file on windows?

**Reply by ayim** [reply]
What steps in the workaround have you done so far and where are you stuck?
Commenting on the Stack Overflow issue for assistance may be your best bet

**Reply by 0dayoverflow** [reply]
where do i add it as a system variable. What text or program do i open?

**Comment by ayim** [comment]
Adding user's log's from https://github.com/orgs/community/discussions/120781

Body
I am working with visual studio 2022 and when I write code I don't get the predictions and in the terminal I get this : Copilot 1.156.0.0 (v1.156.0.0@82fafb995)

GitHub Copilot Agent 1.156.0 (1.156.0)
[DEBUG] [agent] [2024-04-22T21:10:28.919Z] Agent service starting
[DEBUG] [agent] [2024-04-22T21:10:29.350Z] Telemetry initialized
Auth Status: OK
Successfully authorized GitHub Copilot.
[DEBUG] [getCompletions] [2024-04-22T21:10:58.819Z] Requesting completion at position 19:11, between "\r\n asd" and "\r\n\r\n".
[INFO] [default] [2024-04-22T21:10:59.384Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[DEBUG] [getCompletions] [2024-04-22T21:10:59.924Z] Requesting completion at position 19:13, between "\r\n async" and "\r\n\r\n".
[INFO] [ghostText] [2024-04-22T21:11:00.005Z] Cancelled during debounce
[DEBUG] [getCompletions] [2024-04-22T21:11:00.196Z] Requesting completion at position 19:16, between "\r\n abstract" and "\r\n\r\n".
[INFO] [default] [2024-04-22T21:11:00.286Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
statusNotification: Warning: unable to verify the first certificate
[ERROR] [ghostText] [2024-04-22T21:11:00.392Z] Error on ghost text request: (FetchError) unable to verify the first certificate
[ERROR] [certificates] [2024-04-22T21:11:00.393Z] Your current Copilot license doesn't support proxy connections with custom certificates. Please visit https://gh.io/copilot-network-errors to learn more. Original cause: {"type":"system","_name":"FetchError","code":"UNABLE_TO_VERIFY_LEAF_SIGNATURE"}
[INFO] [RequestProposalsAsync] 0 completions available.
[DEBUG] [getCompletions] [2024-04-22T21:11:00.703Z] Requesting completion at position 19:9, between "\r\n p" and "\r\n\r\n".
[DEBUG] [getCompletions] [2024-04-22T21:11:00.751Z] Requesting completion at position 19:15, between "\r\n partial" and "\r\n\r\n".
[INFO] [ghostText] [2024-04-22T21:11:00.843Z] Cancelled during debounce
[DEBUG] [getCompletions] [2024-04-22T21:11:00.932Z] Requesting completion at position 19:12, between "\r\n prop" and "\r\n\r\n".
[INFO] [ghostText] [2024-04-22T21:11:01.014Z] Cancelled during debounce

Added to a message "Your current Copilot license doesn't support proxy connections with custom certificates" and my copilot license is personal and from github pro that added to the fact that I don't belong to any company. what can be a possible solution? Thanks.
imagen

**Comment by ayim** [comment]
@haliovilla what is your proxy setup?

https://github.com/orgs/community/discussions/120781

I'm experiencing the same issue. The strangest thing of all is that GitHub Copilot works fine in Visual Studio Code and even GitHub Copilot Chat works fine within Visual Studio Community 2022.

**Comment by atiris** [comment]
Abbreviated instructions for **Windows and Visual Studio Community 2022**

![image](https://github.com/community/community/assets/10474618/b18f6041-12e6-42c2-9980-a93db6979bfc)

Open [github](https://github.com) in browser (edge) and click on lock, then select `Connection is secure`.

![image](https://github.com/community/community/assets/10474618/0f1ca1d3-bafd-406d-8497-ae3507cfda97)

Click on `Show certificate` button.

![image](https://github.com/community/community/assets/10474618/1609812e-a146-4099-a130-f3b2c3816061)

On `Details` tab click on `Export...` while github.com is selected.

![image](https://github.com/community/community/assets/10474618/bbf987c8-5fc9-4425-aa23-19d64cfd6b40)

Save it somewhere as `certificate chain` and set extension to `.pem`.

![image](https://github.com/community/community/assets/10474618/9d19bb62-9e7a-45a7-80fe-45218ceb685a)

Press `WIN + I` and open Windows Settings, then type: `variables` and select `Edit the system environment variables`.

![image](https://github.com/community/community/assets/10474618/fc6cf242-8ed0-4d7c-8118-549b5ec696f3)

In `System Properties` window click on `Environment Variables...` button. Then on `Environment Variables` window add `New...` user variable.  
Set Variable name: `NODE_EXTRA_CA_CERTS` and Variable value path to your chain certificate. In my case: `c:\Certs\github-chain.pem`.

![image](https://github.com/community/community/assets/10474618/42844daf-5fd8-4a52-afab-6928068d7f80)

Then click OK and close all settings windows. Restart Visual Studio and check if Copilot works ok.

**Comment by J9527H** [comment]
it works for me.
